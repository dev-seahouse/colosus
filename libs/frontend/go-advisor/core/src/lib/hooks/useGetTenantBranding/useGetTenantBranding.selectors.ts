import { useQuery, useQueryClient } from '@tanstack/react-query';
import { useCallback } from 'react';
import { DEFAULT_BRANDING } from '@bambu/go-core';

import type { GetTenantBrandingData } from './useGetTenantBranding';
import { getTenantBrandingQuery } from './useGetTenantBranding';

/**
 * returns current tenant branding
 */
export const useSelectTenantBrandingQuery = () => {
  return useQuery({
    ...getTenantBrandingQuery(),
    select: useCallback((data: GetTenantBrandingData) => {
      return (
        data ?? {
          brandColor: DEFAULT_BRANDING.brandColor,
          logoUrl: DEFAULT_BRANDING.logoUrl,
          headerBgColor: DEFAULT_BRANDING.headerBgColor,
          tradeName: DEFAULT_BRANDING.tradeName,
        }
      );
    }, []),
  });
};

/**
 * query hook to get current tenant trade name
 */
export const useSelectTenantTradeNameQuery = () => {
  return useQuery({
    ...getTenantBrandingQuery(),
    select: useCallback((data: GetTenantBrandingData) => {
      return data.tradeName;
    }, []),
  });
};

/**
 * hook to update tenant branding data in query
 */
export const useSelectUpdateTenantBrandingDataQuery = () => {
  const queryClient = useQueryClient();

  return (branding: Partial<GetTenantBrandingData>) => {
    queryClient.setQueryData(getTenantBrandingQuery().queryKey, (oldData) =>
      oldData ? { ...oldData, ...branding } : oldData
    );
  };
};

/**
 * hook to invalidate tenant branding data query
 */
export const useSelectRefreshTenantBrandingDataQuery = () => {
  const queryClient = useQueryClient();

  return async () =>
    queryClient.invalidateQueries(getTenantBrandingQuery().queryKey);
};
