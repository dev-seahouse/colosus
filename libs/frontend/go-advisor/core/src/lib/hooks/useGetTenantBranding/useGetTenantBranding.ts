import { useQuery } from '@tanstack/react-query';
import { ConnectAdvisorBrandingApi } from '@bambu/api-client';
import type { ConnectAdvisorGetBrandingResponseDto } from '@bambu/api-client';
import type { QueryClient } from '@tanstack/react-query';

export type GetTenantBrandingData =
  Awaited<ConnectAdvisorGetBrandingResponseDto>;

export const getTenantBrandingQuery = () => ({
  queryKey: ['getTenantBranding'],
  queryFn: async () => {
    const connectAdvisorBrandingApi = new ConnectAdvisorBrandingApi();

    const res = await connectAdvisorBrandingApi.getTenantBranding();

    return res.data;
  },
});

export const getTenantBrandingLoader =
  (queryClient: QueryClient) => async (): Promise<GetTenantBrandingData> => {
    const query = getTenantBrandingQuery();

    // ⬇️ return data or fetch it
    return (
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };

export interface UseGetTenantBrandingOptions {
  initialData?: Partial<GetTenantBrandingData>;
}
export function useGetTenantBranding({
  initialData,
}: UseGetTenantBrandingOptions = {}) {
  return useQuery({
    ...getTenantBrandingQuery(),
    initialData,
  });
}

export default useGetTenantBranding;
