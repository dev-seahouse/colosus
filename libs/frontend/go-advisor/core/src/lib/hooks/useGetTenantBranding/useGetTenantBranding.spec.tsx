import {
  renderHook,
  queryClientWrapper,
  waitFor,
} from '@bambu/react-test-utils';

import useGetTenantBranding from './useGetTenantBranding';

describe('useGetTenantBranding', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(() => useGetTenantBranding(), {
      wrapper: queryClientWrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
