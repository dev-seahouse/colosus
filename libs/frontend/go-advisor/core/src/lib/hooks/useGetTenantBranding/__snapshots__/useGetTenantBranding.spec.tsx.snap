// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetTenantBranding > should render successfully 1`] = `
{
  "brandColor": "#00876A",
  "headerBgColor": "#fff",
  "logoUrl": null,
  "tradeName": "Wealth Avenue",
}
`;
